### ISSUE TITLE
FileTransform@1: [Feature Request] JSON case-insensitive substitution

### ISSUE THREAD

**Issue by aytimothy** [microsoft/azure-pipelines-tasks#15213]
## Required Information

Entering this information will route you directly to the right team and expedite traction.

**Question, Bug, or Feature?**  
*Type*: Feature

**Enter Task Name**: FileTransform@1

## Environment
Hosted Agent, though I'm pretty sure this is environment agnostic.

## Issue Description

Is it possible to have JSON file substitution be case-insensitive (or at least add a flag for this)?

We have two sets of JSON configuration files; one for a SPA and another for an ASP.NET Core app.  
In ASP.NET, names are usually upper-camel-case (ie. `ConnectionStrings.Default`), while in React, they are lower-camel-case (ie. `connectionStrings.default`).

Is it possible to add a flag that ignores capitalization as variables with different capitalization are treated as the same (the one defined last takes precedence as the final used value but the one defined first takes precedence in the name) in an Azure Pipeline YML?

### Task logs

We had two jsons, one with:

```json
{
    "connectionStrings": {
        "default": "connectionStrings.default"
    },
    "localization": {
        "en": "localization.en"
    }
}
```

and

```json
{
    "ConnectionStrings": {
        "Default": "ConnectionStrings.Default"
    },
    "Localization": {
        "EN": "Localization.EN"
    }
}
```

and our pipeline had the variables `connectionStrings.default` and `Localization.EN`. In this case, we'll set them to `I GOT SUBSTITUTED!`, but usually a password or localization string goes in its place.

In case, the YML is:

```yml
# Starter pipeline
# Start with a minimal pipeline that you can customize to build and deploy your code.
# Add steps that build, run tests, deploy, and more:
# https://aka.ms/yaml

trigger:
- main

pool:
  vmImage: ubuntu-latest

steps:
- task: FileTransform@1
  inputs:
    folderPath: '$(System.DefaultWorkingDirectory)'
    fileType: 'json'
    targetFiles: '**.json'
- task: PublishBuildArtifacts@1
  inputs:
    PathtoPublish: '$(System.DefaultWorkingDirectory)'
    ArtifactName: 'results'
    publishLocation: 'Container'
```

![image](https://user-images.githubusercontent.com/7418834/130718679-49af86b5-411c-4f0b-b7d3-450a1cf1942c.png)

and the results were incorrect, but as expected:

```json
{
    "connectionStrings": {
        "default": "connectionStrings.default"
    },
    "localization": {
        "en": "I GOT SUBSTITUTED!"
    }
}
```

```json
{
    "ConnectionStrings": {
        "Default": "I GOT SUBSTITUTED!"
    },
    "Localization": {
        "EN": "Localization.EN"
    }
}
```

You can see the repo here: https://dev.azure.com/timothychew0516/Test-Public/_git/case-insensitive-substitution

In case it wasn't clear, is it possible to get a feature that would render both variables in both files be "I GOT SUBSTITUTED"?  
Basically this:

```json
{
    "connectionStrings": {
        "default": "I GOT SUBSTITUTED!"
    },
    "localization": {
        "en": "I GOT SUBSTITUTED!"
    }
}
```

```json
{
    "ConnectionStrings": {
        "Default": "I GOT SUBSTITUTED!"
    },
    "Localization": {
        "EN": "I GOT SUBSTITUTED!"
    }
}
```

## Troubleshooting

There isn't an error since it's "working as expected".

### Error logs

There isn't an error since it's "working as expected".

**Comment by aytimothy** [comment]
Hello,

Has there been any updates? Is it in progress or just backlogged?
